File: Architecture.Web/ClientApp/src/app/Shared/Entity/Common/auditable.ts
import { Entity } from './entity';
import { RecordStatus } from "./record-status";

export class Auditable extends Entity {
    created: Date = new Date();
    modified: Date = new Date();
    createdBy: string = '';
    modifiedBy: string = '';
    recordStatusId: number = 1;
    recordStatus: RecordStatus;

    //clear() {
    //    super.clear();
    //    this.created = null;
    //    this.modified = null;
    //    this.createdBy = '';
    //    this.modifiedBy = '';
    //}
}

File: Architecture.Core/Entities/Core/Auditable.cs
using System;
using Architecture.Core.Entities.LU;

namespace Architecture.Core.Entities.Core
{
    public abstract class Auditable
    {
        public Guid? CreatedBy { get; set; }
        public DateTime Created { get; set; }

        public Guid? ModifiedBy { get; set; }
        public DateTime Modified { get; set; }
        public int? RecordStatusId { get; set; }
        public RecordStatus RecordStatus { get; set; }
    }
}
